src/client/core/request-barrier.js

```diff
@@ -30,46 +30,61 @@ export default class XhrBarrier extends EventEmitter {
     }
 
     _init () {
-        var onXhrSend      = e => this._onXhrSend(e.xhr);
-        var onXhrCompleted = e => this._onXhrCompleted(e.xhr);
-        var onXhrError     = e => this._onXhrError(e.xhr, e.err);
+        var onXhrSend      = e => this._onRequestSend(e.xhr);
+        var onXhrCompleted = e => this._onRequestCompleted(e.xhr);
+        var onXhrError     = e => this._onRequestError(e.xhr, e.err);
+        var onFetchSend    = e => this._onFetchSend(e);
 
         hammerhead.on(hammerhead.EVENTS.xhrSend, onXhrSend);
         hammerhead.on(hammerhead.EVENTS.xhrCompleted, onXhrCompleted);
         hammerhead.on(hammerhead.EVENTS.xhrError, onXhrError);
+        hammerhead.on(hammerhead.EVENTS.fetchSend, onFetchSend);
 
         this._unbindHandlers = () => {
             hammerhead.off(hammerhead.EVENTS.xhrSend, onXhrSend);
             hammerhead.off(hammerhead.EVENTS.xhrCompleted, onXhrCompleted);
             hammerhead.off(hammerhead.EVENTS.xhrError, onXhrError);
+            hammerhead.off(hammerhead.EVENTS.fetchSend, onFetchSend);
         };
     }
 
-    _onXhrSend (xhr) {
+    _onRequestSend (request) {
```

Comment:
I suppose we should leave the old name for these functions. They handle just xhrs events, so it's `_onXhrSend/Completed/Error`.
In our context `request` it's xhr+fetch